### DISCUSSION TITLE
pyodbc.connect() gives an error when executing it with pytest

### DISCUSSION THREAD

**Discussion by javierhersan** [mkleehammer/pyodbc#1353] [Q&A]
When executing the following code inside of my test function

```
conn = pyodbc.connect(connection_string)
```

It throws the following error:

```
pyodbc.Error: ('HY000', 'The driver did not supply an error!')
```

The connection string of my tests is the same as in my project. 

```
connection_string = 'DRIVER={ODBC Driver 18 for SQL Server};SERVER=tcp:database.windows.net;PORT=1433;DATABASE=database;UID=admin;PWD=pass;'
```

The test project also uses the same virtual environment with the same dependencies.

What happens here?

**Comment by v-chojas** [comment]
You can start by providing more information about your environment, and preferably an ODBC trace too.

**Reply by gordthompson** [reply]
question also posted on Stack Overflow [here](https://stackoverflow.com/q/78433094/2144390)
